drape_frontend/rule_drawer.cpp

```diff
@@ -457,27 +413,25 @@ void RuleDrawer::operator()(FeatureType & f)
        !ftypes::IsBuildingChecker::Instance()(types)))
     return;
 
+  if (ftypes::IsCoastlineChecker::Instance()(types) && !CheckCoastlines(f))
+    return;
+
   Stylist const s(f, m_zoomLevel, m_deviceLang);
 
   // No drawing rules.
-  if (s.m_symbolRule == nullptr && s.m_captionRule == nullptr && s.m_houseNumberRule == nullptr &&
-      s.m_lineRules.empty() && s.m_areaRule == nullptr && s.m_hatchingRule == nullptr)
-    return;
-
-  if (s.m_isCoastline && !CheckCoastlines(f))
+  if (!s.m_symbolRule && !s.m_captionRule && !s.m_houseNumberRule &&
+      s.m_lineRules.empty() && !s.m_areaRule && !s.m_hatchingRule)
     return;
 
 #ifdef DEBUG
   // Validate mixing of feature styles.
   bool const hasLine = !s.m_lineRules.empty();
-  bool const hasLineAdd = s.m_shieldRule != nullptr || s.m_pathtextRule != nullptr;
-  bool const hasPoint = s.m_symbolRule != nullptr || s.m_captionRule != nullptr || s.m_houseNumberRule != nullptr;
-  bool const hasArea = s.m_areaRule != nullptr || s.m_hatchingRule != nullptr;
+  bool const hasLineAdd = s.m_shieldRule || s.m_pathtextRule;
+  bool const hasPoint = s.m_symbolRule || s.m_captionRule || s.m_houseNumberRule;
```

Comment:
Can you reuse these vars in the if above to avoid double checking of the same flags?

Reply:
nope, this code is under `#ifdef DEBUG`